src/daemon.rs

```diff
@@ -445,6 +454,79 @@ impl DaemonContext {
         }
         Ok(())
     }
+
+    fn dht_key_to_fps(&self, key: &str) -> Vec<Fingerprint> {
+        let mut all_nodes: Vec<Fingerprint> = self.relay_graph.read().all_nodes().collect();
+        all_nodes.sort_unstable_by_key(|fp| *blake3::hash(&(key, fp).stdcode()).as_bytes());
+        all_nodes
+    }
+
+    pub async fn dht_insert(&self, key: Fingerprint, value: HavenLocator) {
+        let replicas = self.dht_key_to_fps(&key.to_string());
+
+        for replica in replicas.into_iter().take(DHT_REDUNDANCY) {
+            log::debug!("key {key} inserting into remote replica {replica}");
+            let gclient = GlobalRpcClient(GlobalRpcTransport::new(self.clone(), replica));
+            match gclient
+                .dht_insert(key, value.clone(), false)
+                .timeout(Duration::from_secs(10))
+                .await
+            {
+                Some(Err(e)) => log::debug!("inserting {key} into {replica} failed: {:?}", e),
+                None => log::debug!("inserting {key} into {replica} timed out"),
+                _ => {}
+            }
+        }
+    }
+
+    pub async fn dht_get(&self, key: Fingerprint) -> Option<HavenLocator> {
```

Comment:
Just do the signature verification in dht_get and get rid of get_rendezvous and insert_rendezvous.